rollup-interface/src/state_machine/da.rs

```diff
@@ -68,6 +68,27 @@ pub trait DaVerifier {
     ) -> Result<Self::ValidityCondition, Self::Error>;
 }
 
+/// [`AccumulatorStatus`] is a wrapper around an accumulator vector that specifies
+/// whether a [`CountedBufReader`] has finished reading the underlying buffer.
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum AccumulatorStatus {
```

Comment:
I would like to suggest not owned wrapper, but referenced:

```rust
pub enum AccumulatorStatus<'a> {
    Completed(&'a[u8]),
    InProgress(&'a[u8])
}
```

and remove `pub inner_vec`, so callers have to deal with matching themself.

Because API with `inner_vec` can be misused easily, when dev can always call it like that:

```
let i_need_data_now = counted_buf_reader.accumulator().inner_vec()
```

And then it just does not care if it was read or not.

Reply:
Oh, now I see that `inner_vec` is used by CountedBufReader itself.

I think that `CountedBufReader` can maintain internal vec as it was doing before, just forbid direct access to it, only via `AccumulatorStatus` (which can be just renamed to`Accumulator`)